pkg/recipes/engine/engine.go

```diff
@@ -184,16 +184,26 @@ func (e *engine) GetRecipeMetadata(ctx context.Context, recipeDefinition recipes
 
 // getRecipeMetadataCore function is the core logic of the GetRecipeMetadata function.
 // Any changes to the core logic of the GetRecipeMetadata function should be made here.
-func (e *engine) getRecipeMetadataCore(ctx context.Context, recipeDefinition recipes.EnvironmentDefinition) (map[string]any, error) {
+func (e *engine) getRecipeMetadataCore(ctx context.Context, opts GetRecipeMetadataOptions) (map[string]any, error) {
+	// Load environment configuration to get the recipe config information which contains the secrets.
+	configuration, err := e.options.ConfigurationLoader.LoadConfiguration(ctx, opts.Recipe)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Could you add context in the comment that this is needed to download a recipe from private registries, specifically Terraform recipes currently? It wasn't obvious from the code that why are secrets needed here.

Reply:
updated the comment